### ISSUE TITLE
RuntimeInstanceNotAllowed Error Initializing XmlSerializer on Android Release Build in .NET 9

### ISSUE THREAD

**Issue by david-maw** [dotnet/maui#25789]
### Description

App fails with an exception in a release build of .NET 9 (.NET 8 and debug builds of .NET 9 work fine).

It looks like the fault happened because of a call on `new XmlSerializer(typeof(Meal))` (deferring that call defers the fault).

The abbreviated stack trace looks like this:
```
System.ArgumentException: RuntimeInstanceNotAllowed
  ?, in object DefaultValueAttribute.get_Value()
  ?, in new XmlAttributes(ICustomAttributeProvider)
  ?, in XmlAttributes XmlReflectionImporter.GetAttributes(MemberInfo)
  ?, in bool XmlReflectionImporter.InitializeStructMembers(StructMapping, StructModel, bool, string, RecursionLimiter)
  ?, in StructMapping XmlReflectionImporter.ImportStructLikeMapping(StructModel, string, bool, XmlAttributes, RecursionLimiter)
...
(1 additional frame(s) were not displayed)

System.InvalidOperationException: XmlTypeReflectionError, DivisiBill.Models.Meal
  ?, in XmlSerializer Meal.get_MealSerializer()
  ?, in void Meal.SaveToStream(Stream streamParameter)
  ?, in void Meal.SaveToApp()
  ?, in void Meal.OverwriteCurrent()
  ?, in async Task Meal.BecomeCurrentMealAsync()
...
(12 additional frame(s) were not displayed)
```

### Steps to Reproduce

1. Clone the repository, pull the `class-constructor-fault` branch.
2. Start VS2022 and select the DivisiBill project and a release build.
3. At the VS developer PowerShell prompt enter `dotnet restore`
4. Run the app without debugging on an Android Emulator or physical device (ctrl+F5). You should see:![Image](https://github.com/user-attachments/assets/08a04044-9758-42bd-a710-6e909bd0b85a)
5.Then![Image](https://github.com/user-attachments/assets/c3b7fdb9-b7ac-4546-8d7c-9c7196f8c699)
6.Press the left arrow to see:![Image](https://github.com/user-attachments/assets/4a0955e7-be47-4c02-bc5d-cbdc2240cdbe)
7. Click "While Using the App" and see:![Image](https://github.com/user-attachments/assets/62468a28-3d83-4553-a4c7-a236d6cdd44a)
8. At this point the app will fault.

If it were working correctlyyou would see:![Image](https://github.com/user-attachments/assets/49d7ab55-8027-4cc5-870d-913639d1c815)
Then![Image](https://github.com/user-attachments/assets/408192e6-4af0-4e63-90e0-04c9a56992f2)
 

 

### Link to public reproduction project repository

https://github.com/david-maw/DivisiBill.git

### Version with bug

9.0.0-rc.2.24503.2

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.93 SR9.3

### Affected platforms

Android

### Affected platform versions

Android 14

### Relevant log output

```shell

```

**Comment by drasticactions** [comment]
I don't believe your issue is related to the MAUI UI framework, it's a runtime one:

https://github.com/dotnet/runtime/blob/release/8.0/src/libraries/System.Private.CoreLib/src/System/ComponentModel/DefaultValueAttribute.cs

https://github.com/dotnet/runtime/blob/release/9.0/src/libraries/System.Private.CoreLib/src/System/ComponentModel/DefaultValueAttribute.cs#L246-L257

Between 8.0 and 9.0, a switch was added (https://github.com/dotnet/runtime/pull/100416 and https://github.com/dotnet/runtime/pull/105766) to throw on these values. Most likely what's tripping it up on Release mode is trimming (which would explain it working in Debug mode, and not Release.) I'm not sure how to specifically handle trimming with XmlSerializer, @jonathanpeppers would you know of what to do here?
